core/src/main/java/org/apache/calcite/sql/SqlLiteral.java

```diff
@@ -284,6 +284,11 @@ public static Comparable value(SqlNode node) {
         return (NlsString) literal.value;
       case NUMERIC:
         return (BigDecimal) literal.value;
+      case INTERVAL_YEAR_MONTH:
+      case INTERVAL_DAY_TIME:
+        final SqlIntervalLiteral.IntervalValue val =
+            (SqlIntervalLiteral.IntervalValue) literal.value;
+        return val.getSign();
```

Comment:
>To make the monotonicity correct, we need to let SqlLiteral.value() (line: 291) return -1.
>Make sense?

It does not.
You are changing `value` method. It's javadoc says `* Extracts the value from a literal.`.
It's javadoc explicitly says  `If the node is a {@link SqlIntervalQualifier...` and you are returning just sign.

That does not work.